src/time-picker/TimePicker.js

```diff
@@ -0,0 +1,226 @@
+  const inputArray = cleanedInput.split(separator);
+  const [_hours, _minutes] = inputArray;
+  const hours = numberRegex.test(_hours) ? parseInt(_hours, 10) : '';
+  const minutes = !_minutes || numberRegex.test(_minutes) ? parseInt(_minutes, 10) : '';
+  const validHours = isValidHoursInTimeFormat(hours, timeFormat);
+  const validMinutes = Minute.is(minutes);
+
+  if (validHours && validMinutes) {
+    return { hours, minutes, originalInput: cleanedInput };
+  } else if (validHours) {
+    return { originalInput: cleanedInput };
+  } else {
+    return inputError;
+  }
+};
+
+export const createTimeList = ({ hours, minutes }, timeFormat) => {
+  if (!isValidHoursInTimeFormat(hours) || !Minute.is(minutes)) {
+    const hoursList = range(0, 24);
+    const minutesList = range(0, 60, interval);
+    return flatten(hoursList.map(hours => minutesList.map(minutes => ({ hours, minutes, timeFormat })
+    )));
+  } else {
+    return timeFormat !== H24 ? [
+      { hours, minutes, timeFormat },
+      { hours: hoursInH12(hours), minutes, timeFormat }
+    ] : [
+      { hours, minutes, timeFormat }
+    ];
+  }
+};
+
+const startsWith = (time, originalInput) => {
+  const timeStr = formatTime(time);
+  return timeStr.substr(0, originalInput.length) === originalInput;
+};
+
+export const filterTime = ({ originalInput, minTime, maxTime }) => time => {
+  return lteTime(minTime, time) && lteTime(time, maxTime) && startsWith(time, originalInput);
+};
+
+// we are not "filtering" options (options array is always empty and discarded by this function)
+// our `filterOptions` actually generates options based on current input string and props.
+// We use this as a workaround because there's not any other easy way of updating options
+// without branking input in react-select v0.6.x
+// NOTE: can be fixed updating to v1.0.0
+export const makeFilterOptions = ({ minTime, maxTime, timeFormat }) => (_, inputStr) => {
+  const time = parseInTimeFormat(inputStr, timeFormat);
+  const timeList = time === inputError ? [] : createTimeList(time, timeFormat);
+  const filteredTimeList = timeList.filter(filterTime({
+    originalInput: time.originalInput, minTime, maxTime
+  }));
+  return filteredTimeList.map(toOption);
+};
+
+@skinnable()
+@props(Props)
+export default class TimePicker extends React.Component {
+
+  static defaultProps = {
+    placeholder: `--${separator}--`,
+    timeFormat: H24,
+    minTime: { hours: 0, minutes: 0 },
+    maxTime: { hours: 23, minutes: 59 }
+  }
+
+  _onChange = (value) => {
+    if (value) {
+      // interface with component user is always in H24
+      const time = parseInTimeFormat(value, H24);
+      this.props.onChange(time);
+    } else {
+      this.props.onChange();
+    }
+  };
+
+  getLocals() {
+    const {
+      timeFormat,
+      className,
+      value: userValue,
+      ...props
+    } = this.props;
+
+    const value = userValue ? toOption({
+      ...userValue, timeFormat: this.props.timeFormat
```

Comment:
you already has the variable `timeFormat` in scope. You can safely replace `timeFormat: this.props.timeFormat` with just `timeFormat`